x-pack/plugins/ml/server/models/results_service/results_service.ts

```diff
@@ -605,6 +612,104 @@ export function resultsServiceProvider(mlClient: MlClient) {
+    start,
+    end,
+  }: DatafeedResultsChartDataParams): Promise<GetDatafeedResultsChartDataResult> {
+    const finalResults: GetDatafeedResultsChartDataResult = {
+      bucketResults: [],
+      datafeedResults: [],
+    };
+
+    const { getDatafeedByJobId } = datafeedsProvider(client!, mlClient);
+    const datafeedConfig = await getDatafeedByJobId(jobId);
+
+    const {
+      body: { jobs },
+    } = await mlClient.getJobs({ job_id: jobId });
+    const jobConfig = jobs[0];
+    const timefield = jobConfig.data_description.time_field;
+    const bucketSpan = jobConfig.analysis_config.bucket_span;
+
+    if (datafeedConfig === undefined) {
+      return finalResults;
+    }
+
+    const rangeFilter = {
+      range: {
+        [timefield]: { gte: start, lte: end },
+      },
+    };
+
+    let datafeedQueryClone =
+      datafeedConfig.query !== undefined ? cloneDeep(datafeedConfig.query) : defaultSearchQuery;
+
+    if (datafeedQueryClone.bool !== undefined) {
+      if (datafeedQueryClone.bool.filter === undefined) {
+        datafeedQueryClone.bool.filter = [];
+      }
+      if (Array.isArray(datafeedQueryClone.bool.filter)) {
+        datafeedQueryClone.bool.filter.push(rangeFilter);
+      } else {
+        // filter is an object so convert to array so we can add the rangeFilter
+        const filterQuery = cloneDeep(datafeedQueryClone.bool.filter);
+        datafeedQueryClone.bool.filter = [filterQuery, rangeFilter];
+      }
+    } else {
+      // Not a bool query so convert to a bool query so we can add the range filter
+      datafeedQueryClone = { bool: { must: [datafeedQueryClone], filter: [rangeFilter] } };
+    }
+
+    const esSearchRequest = {
+      index: datafeedConfig.indices.join(','),
+      body: {
+        query: datafeedQueryClone,
+        ...(datafeedConfig.runtime_mappings
+          ? { runtime_mappings: datafeedConfig.runtime_mappings }
+          : {}),
+        aggs: {
+          doc_count_by_bucket_span: {
+            date_histogram: {
+              field: timefield,
+              fixed_interval: bucketSpan,
+            },
+          },
+        },
+        size: 0,
+      },
+      // @ts-expect-error @elastic/elasticsearch Datafeed is missing indices_options
+      ...(datafeedConfig?.indices_options ?? {}),
+    };
+
+    if (client) {
+      const {
+        body: { aggregations },
+      } = await client.asCurrentUser.search(esSearchRequest);
+
+      finalResults.datafeedResults =
+        // @ts-expect-error incorrect search response type
+        aggregations?.doc_count_by_bucket_span?.buckets.map((result) => [
+          result.key,
+          result.doc_count,
+        ]) || [];
+    }
+
+    const bucketResp = await mlClient.getBuckets({
+      job_id: jobId,
+      body: { desc: true, start: String(start), end: String(end), page: { from: 0, size: 1000 } },
+    });
+
+    const bucketResults = get(bucketResp, ['body', 'buckets'], []);
+    each(bucketResults, (dataForTime) => {
```

Comment:
Similarly, I don't see the benefit of using lodash's `each` here, `bucketResults.forEach` would be better IMO

Reply:
Removed all lodash function usage in 1df1594527f9d947920775b41019c80641ca2777